### ISSUE TITLE
[F12 GTD Source Link] F12 does not fall back to decompilation if Source Link/Embedded file lookup has exception.

### ISSUE THREAD

**Issue by chuckries** [dotnet/roslyn#69371]
**Version Used**: 
Microsoft Visual Studio Enterprise 2022 (64-bit) - Int Preview
Version 17.8.0 Preview 2.0 [34002.499.main]

**Steps to Reproduce**:
Filing this issue on in response to this [Visual Studio Feedback Item](https://developercommunity.visualstudio.com/t/Go-To-Definition-stops-working-for-Sourc/10428087).

1. In VS 2022, make a new C# Console App targeting net6.0 or above. 
2. Add a PackageReference to Avalonia 11.0.2
```
  <ItemGroup>
    <PackageReference Include="Avalonia" Version="11.0.2" />
  </ItemGroup>

```
3. Add a code reference to `Avalonia.Controls.Templates.IDataTemplate`
```cs
Avalonia.Controls.Templates.IDataTemplate foo = null;
```
4. Place caret on `IDataTemplate`
5. F12

**Expected Behavior**:
Due to a [malformed embedded PDB in Avalonia](https://github.com/AvaloniaUI/Avalonia/issues/12431), consuming their embedded pdb will cause BadImageFormatException. At that point, I would expect Roslyn to fall back to decompiling the file and showing the decompiled or Metadata only file for F12.

**Actual Behavior**:
Nothing happens, F12 goes nowhere.

**Comment by chuckries** [comment]
cc @davidwengier @tmat

**Comment by CyrusNajmabadi** [comment]
Honestly, I'm ok with the behavior here. If it's malformed then any behavior seems fine to me (including blacklisting the dll and offering no support for it).  Whoever is forming this wrong should fix this on their end. Otherwise we're just investing and maintaining code to workaround bugs elsewhere.

**Comment by chuckries** [comment]
As mentioned [here](https://github.com/AvaloniaUI/Avalonia/pull/12410#issue-1831654386), Rider falls back to decompilation in this case. I don't see why we wouldn't put a log line in the Navigate To Source output pane stating that the located PDB is poorly formatted, and then decompile the code. In this case, the dll is not malformed, just the PDB; the dll could still be decompiled.

**Comment by chuckries** [comment]
If nothing else, we need to log the fact that there was an error, which is not currently happening.

**Comment by CyrusNajmabadi** [comment]
> I don't see why we wouldn't put a log line in the Navigate To Source output pane 

Is tehre such an output pane?  I'm fine logging some sort of message.  

> In this case, the dll is not malformed, just the PDB;

This still seems to be a problem with that project.  Why are they producing malformed PDBs?  Why is tehre an expectation that any downstream consumers have to workaround this?  I would just have them fix their core issue, and then users will get an actually good experience. 

In other words, i'd rather treat the root cause, versus a symptom of that strange situation.  Otherwise, bugs from other teams just end up as taxes on downstream projects.

**Comment by Mrxx99** [comment]
The issue with pdbs was fixed in Avalonia 11.0.3

**Comment by CyrusNajmabadi** [comment]
my preference then is to close this out.  Users would only see this in the event of a very abnormal situation.  And that abnormal situation has now been rectified.
